test/llvm-intrinsics/memmove.ll

```diff
@@ -81,6 +95,12 @@ define spir_func void @copy_struct(%struct.SomeStruct addrspace(1)* nocapture re
   ret void
 }
 
+; Function Attrs: nounwind
+define spir_kernel void @test_array(i8 addrspace(1)* %in, i8 addrspace(1)* %out) {
```

Comment:
Here is what we get with your fix after SPIRVLowerMemmove pass:
```
define spir_kernel void @test_array(i8 addrspace(1)* %in, i8 addrspace(1)* %out) {
  %1 = alloca i8, align 1
  call void @llvm.lifetime.start.p0i8(i64 -1, i8* %1)
  call void @llvm.memcpy.p0i8.p1i8.i32(i8* %1, i8 addrspace(1)* %out, i32 128, i1 false)
  call void @llvm.memcpy.p1i8.p0i8.i32(i8 addrspace(1)* %in, i8* %1, i32 128, i1 false)
  call void @llvm.lifetime.end.p0i8(i64 -1, i8* %1)
  ret void
}
```
We allocate 8 bits and copy 128*32(=4096) bit there. I believe we should see a segmentation fault (or something like this ) in this case.

Reply:
I see. I believe we need to have:
`%1 = alloca i8*`
instead: https://godbolt.org/z/EYY3ezacf